examples/numgen.py

```diff
@@ -83,52 +82,62 @@ def __init__(self, name: str, table: str = "number", parent: Optional[QObject] =
         self.generatorFrame.dbBox.addItem("")
         self.viewerFrame = ViewerFrame()
         # connect signals to slots
-        self.received.connect(self.updateDB)
         self.generatorFrame.dbBox.currentIndexChanged.connect(self.setDB)
         self.generatorFrame.generateButton.clicked.connect(self.generateNumber)
 
     def frames(self) -> Tuple[GeneratorFrame, ViewerFrame]:
         """Overridden."""
         return (self.generatorFrame, self.viewerFrame)
 
-    @pyqtSlot(str, str)
-    def updateDB(self, channelName: str, msg: str):
+    def updateDB(self, content: dict):
         """Updates the database list using the transferred message.
 
-        This is a slot for received signal.
         It assumes that:
             The new database is always added at the end.
             Changing the order of the databases is not allowed.
 
         Args:
-            channelName: A name of the channel that transfered the signal.
-            msg: An input message to be transferred through the channel.
+            channelName: Channel name that transferred the message.
+            content: Received content.
               The structure follows the message protocol of DBMgrApp.
         """
+        originalDBs = set(self.dbs)
+        newDBs = set([""])
+        for db in content.get("db", ()):
+            if any(key not in db for key in ("name", "path")):
+                print(f"The message was ignored because "
+                        f"the database {db} has no such key; name or path.")
+                continue
+            name, path = db["name"], db["path"]
+            newDBs.add(name)
+            if name not in self.dbs:
+                self.dbs[name] = path
+                self.generatorFrame.dbBox.addItem(name)
+        removingDBs = originalDBs - newDBs
+        if self.generatorFrame.dbBox.currentText() in removingDBs:
+            self.generatorFrame.dbBox.setCurrentText("")
+        for name in removingDBs:
+            self.dbs.pop(name)
+            self.generatorFrame.dbBox.removeItem(self.generatorFrame.dbBox.findText(name))
+
+    def receivedSlot(self, channelName: str, content: Any):
+        """Handles the received broadcast message.
```

Comment:
Please use the notation agreed in #44.

Reply:
Oh, I forgot it. Thanks for letting me know.